components/net/http_cache.rs

```diff
@@ -474,11 +498,22 @@ fn handle_range_request(
                 } else {
                     continue;
                 };
-                if res_beginning < beginning && res_end == total - 1 {
+                let total_minus_one = {
+                    match total.checked_sub(1) {
+                        Some(minus_one) => minus_one,
+                        _ => continue,
+                    }
+                };
+                if res_beginning < beginning && res_end == total_minus_one {
                     let resource_body = &*partial_resource.body.lock().unwrap();
                     let requested = match resource_body {
                         &ResponseBody::Done(ref body) => {
-                            let from_byte = beginning as usize - res_beginning as usize;
+                            let from_byte = {
+                                match beginning.checked_sub(res_beginning) {
```

Comment:
Since we have established that `res_beginning < beginning` there is no need to check this operation.